CHANGELOG.md

```diff
@@ -1,3 +1,255 @@
+- `aws-ebs-csi-driver`: Add dynamic provisioning, resizing and snapshot options ([#1277](https://github.com/kinvolk/lokomotive/pull/1277)). Now the user has the ability to control the AWS EBS driver to enable or disable provisioning, resizing and snapshotting.
+
+#### Security enhancements
+
+- `calico-host-protection`: Add custom locked down PSP configuration ([#1274](https://github.com/kinvolk/lokomotive/pull/1274)).
+
+#### Documentation
+
+- Add `openebs-operator` update guide ([#1163](https://github.com/kinvolk/lokomotive/pull/1163)).
+- Add `rook-ceph` update guide ([#1165](https://github.com/kinvolk/lokomotive/pull/1165)).
+
+#### Miscellaneous
+
+- Pull control plane images from Quay to avoid hitting Docker Hub pulling limits ([#1226](https://github.com/kinvolk/lokomotive/pull/1226)).
+- Bootkube now waits for all control plane charts to converge before exiting, which should make the bootstrapping process more stable ([#1085](https://github.com/kinvolk/lokomotive/pull/1085)).
+- Remove deprecated CoreOS mentions from AWS ([#1245](https://github.com/kinvolk/lokomotive/pull/1245)) and bare metal ([#1246](https://github.com/kinvolk/lokomotive/pull/1246)).
+- Improve hardware reservations validation rules on Equinix Metal ([#1186](https://github.com/kinvolk/lokomotive/pull/1186)).
+
+### Updating from v0.5.0
+
+#### Configuration syntax changes
+
+##### AWS
+
+Removed the undocumented `cluster.os_name` parameter, since Lokomotive supports Flatcar Container Linux only.
+
+##### Bare-metal
+
+The `cluster.os_channel` parameter got simplified by removing the `flatcar-` prefix.
+
+###### Old
+
+```hcl
+os_channel = "flatcar-stable"
+```
+
+###### New
+
+```hcl
+os_channel = "stable"
+```
+
+##### Velero
+
+Velero requires an explicit `provider` field to select the provider.
+Example:
+
+```hcl
+component `velero` {
+  provider = "openebs"
+
+  openebs {
+    ...
+  }
+}
+```
+
+#### Cluster update steps
+
+> **NOTE**: Updating multiple Lokomotive versions at a time is not supported. If your cluster is running a version older than `v0.5.0`, update to `v0.5.0` first and only then proceed with the update to `v0.6.0`.
+
+Please perform the following manual steps in your cluster configuration directory.
+
+1. Download the release bundle.
+
+  ```bash
+  curl -LO https://github.com/kinvolk/lokomotive/archive/v0.6.0.tar.gz
+  tar -xvzf v0.6.0.tar.gz
+  ```
+
+2. Install the Packet CCM.
+
+  If you are running Lokomotive on Equinix Metal (formerly Packet), then install Packet CCM. Export your Packet cluster's project ID and API Key.
+
+  ```bash
+  export PACKET_API_KEY=""
+  export PACKET_PROJECT_ID=""
+
+  echo "apiKey: $PACKET_API_KEY
+projectID: $PACKET_PROJECT_ID" > /tmp/ccm-values.yaml
+
+  helm install packet-ccm --namespace kube-system --values=/tmp/ccm-values.yaml ./lokomotive-0.6.0/assets/charts/control-plane/packet-ccm/
```

Comment:
I noticed the following error message from the CCM

```
E0111 15:08:09.567379       1 node_lifecycle_controller.go:154] error checking if node test-iago-pool-1-worker-0 is shutdown: providerID cannot be empty string
```

I guess the CCM couldn't populate the providerID in the node object because it's immutable. If we wanted to have it we'd need to recreate the node object.

I'm not sure if this causes problems, can you have a look?

Reply:
I could not reproduce that error @iaguis `providerID cannot be empty string`.

Reply:
>I guess the CCM couldn't populate the providerID in the node object because it's immutable. If we wanted to have it we'd need to recreate the node object.

CCM does not populate providerID IIRC. It must be done by kubelet. And it will be done when you restart kubelet service (restarting should de-register the node).

Reply:
> I could not reproduce that error @iaguis `providerID cannot be empty string`.

FTR this was updating from v0.5.0. I guess this is what you were trying but wanted to mention it anyway.

> CCM does not populate providerID IIRC. It must be done by kubelet. And it will be done when you restart kubelet service (restarting should de-register the node).

Hm, I wonder how this happened for me then...

Reply:
> FTR this was updating from v0.5.0. I guess this is what you were trying but wanted to mention it anyway.

Yep I did test it by upgrading from v0.5.0.